Unified testsuite (#4334)
ComputationalRadiationPhysics/picongpu: docs/source/testing/testbuilding.rst

    
These parameters are all optional.
They serve to provide a better overview and a better presentation of the results after the test.
As already mentioned, there are two variants of documenting the test results, the text format as a .log file and a plot.
The author is only mentioned in the .log file, while the title is used as a heading for the test.
If the plot_title parameter is not set or is defined as None, the title is used as the plot title.

2. paths to files
"""""""""""""""""

.. code-block:: python

    # directory specified
    resultDirection = None
    paramDirection = None
    dataDirection = None
    
In order to be able to evaluate the test, the test suite needs to know what data it can use and where it gets it from.
To ensure this, the paths to the individual folders can be specified here.
If data from these files is required, these parameters must be set unless they are passed to main.py. 
(See the :ref:`main.py documentation <usage>` for this).

``resultDirection`` designates the folder in which the result of the test is to be saved.
If an automatic test is to be carried out and you are interested in the results in detail, this value should be set.

> ``dataDirection`` gives the path to the .txt files of the simulation (note in-situ simulation).

Reviewer: What do you mean by `note in-situ simulation`?

``paramDirection`` and ``jsonDirection`` are available to include .param files and .json files in the evaluation.

The test suite then needs to know which parameters it has to read out. 
The parameters ``data_Parameter``, ``param_Parameter`` and ``json_Parameter`` are available for this.
In the MI example, no ``json_parameter`` is required:

.. code-block:: python

    # parameter information found in .param files
    param_Parameter = ["gamma", "BASE_DENSITY_SI", "DELTA_T_SI"]
    data_Parameter = ["Bz", "step"]
    
3. test condition
"""""""""""""""""

In order for the test suite to be able to decide whether a test passed or not, it needs an acceptance range.
This is done in the form of the parameter ``acceptance``, which contains the maximum deviation.

> An acceptance range of 0.2 thus describes a maximum deviation of 20% in which the test is passed.

Reviewer: ```suggestion
An acceptance range of 0.2 thus describes a maximum deviation of 20% from an expected value in which the test is passed.
```

.. code-block:: python

    # acceptance in percentage
    acceptance = 0.2
    
4. Plot data
""""""""""""

.. code-block:: python

    # plot data
    plot_xlabel = r"$t[\omega_{pe}^ {-1}]$"
    plot_ylabel = r"$\Gamma_\mathrm{Fi}$"
    # if None or not defined the standard type will be used, see documentation
    plot_type = None
    # if None or not defined the time will be used
    plot_xaxis = None
    # for more values see the documentation (e.g. 2D plot needs zaxis and yaxis)
    

In this section, all data required for the plot display are transmitted.
Possible parameters are ``plot_title``, ``plot_xlabel``, ``plot_ylabel``, ``plot_type`` and ``plot_xaxis``.
The first three define the annotation of the plot.

``plot_type`` can assume the values 1D or 2D and describes the dimensionality of the plot.

> If no value is specified, 1D is used by default.

Reviewer: Do you mean x-y plots with 1D?

Reviewer: @MSVoss ping

Author: Yes, thats correct

> ``plot_xaxis`` beschreibt die WErte der x-Achse(default: time).

Reviewer: ```suggestion
``plot_xaxis`` describes the values of the x-axis (default: time).
```
